src/utils/ruleUtils.ts

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import * as vscode from "vscode";
+import { UiStrings } from "../uiStrings";
+
+const rulesetFileKey = "rulesetFile";
+
+export async function setRulesetFile(newRulesetFile: string, alwaysShowSetInfo: boolean = true) {
+    const spectralLinterConfig = vscode.workspace.getConfiguration("spectral");
+    const currentRulesetFile = spectralLinterConfig.get<string>(rulesetFileKey);
+
+    const needToUpdate = newRulesetFile !== currentRulesetFile;
+
+    if (needToUpdate) {
+        await spectralLinterConfig.update("rulesetFile", newRulesetFile, vscode.ConfigurationTarget.Global);
```

Comment:
so if user select the rulesetfile, close the vscode, and reopen the vscode, the rulesetfile may also guide the new project? is this by designed?

Reply:
Yes. If needed, user could always trigger 'Set API Guide' command to select different options.